doc/modules/computing.rst

```diff
@@ -504,20 +504,47 @@ Links
   - `Scipy sparse matrix formats documentation <https://docs.scipy.org/doc/scipy/reference/sparse.html>`_
 
 Parallelism, resource management, and configuration
-=====================================================
+===================================================
 
 .. _parallelism:
 
-Parallel and distributed computing
------------------------------------
+Parallelism
+-----------
 
-Scikit-learn uses the `joblib <https://joblib.readthedocs.io/en/latest/>`__
-library to enable parallel computing inside its estimators. See the
-joblib documentation for the switches to control parallel computing.
+Scikit-learn supports parallel implementations in two ways: via the `joblib
+<https://joblib.readthedocs.io/en/latest/>`_ library, or via OpenMP (through
+Cython, which allows to release the GIL).
+
+When the underlying implementation uses joblib, the number of jobs that are
+spawned in parallel can be controled via the ``n_jobs`` parameter. In most
+cases, scikit-learn uses joblib to rely on multi-processing rather than on
+multi-threading.
```

Comment:
right, also it seems that the default is now loky which prefers using threads anyway.

@ogrisel @jeremiedbb @tomMoral I think it would be useful to have a description here of what users can expect (threads vs processes) when joblib is used, as well as what it means in terms of memory usage.

Could you propose something? It's always been confusing to me TBH.

Reply:
the default is loky. But calling joblib.Parallel with `prefer='threads'` will choose the threading backend, unless the user has specified a backend himself.

```
joblib.Parallel()  # loky (default)

joblib.Parallel(prefer='threads') # threading

with joblib.parallel_backend("loky"):
    joblib.Parallel(prefer="threads")  # loky
```

Reply:
in terms of memory usage, process based parallelism will duplicate a lot of things so you should expect an increase of memory usage. But loky mitigate this a bit by memmaping X if it's more than 1MB.

Reply:
Thanks for the feedback. I really think all this should be clearly documented. I'll try to come up with something.

Reply:
> loky mitigate this a bit by memmaping X if it's more than 1MB.

The mitigation is for all process-based backends (not only `loky`) for any `numpy` array over 1MB.